src/openrct2-ui/interface/ViewportInteraction.cpp

```diff
@@ -348,7 +347,8 @@ int32_t viewport_interaction_get_item_right(int32_t x, int32_t y, viewport_inter
             break;
 
         case VIEWPORT_INTERACTION_ITEM_BANNER:
-            banner = &gBanners[tileElement->AsBanner()->GetIndex()];
+        {
+            auto banner = tileElement->AsBanner()->GetBanner();
```

Comment:
Looks a little bit funny, but I'm guessing thats because both have different storage.

Reply:
The tile elements should be postfixed with Element to make it clearer, so SurfaceTileElement, FootpathTileElement, BannerTileElement, etc.

Reply:
Makes sense, not requesting changes here, just odd looking.